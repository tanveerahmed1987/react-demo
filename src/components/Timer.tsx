import React, { useState, useEffect } from "react";
import styled from "styled-components";

const TimeWrapper = styled.div`
  font-family: Roboto-Bold;
  font-size: 84px;
  color: #37474f;

  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export const Timer: React.FC = () => {
  const [currentTime, setCurrentTime] = useState({ time: new Date() });

  useEffect(() => {
    const interval = setInterval(
      () => setCurrentTime({ time: new Date() }),
      1000
    );
    return () => {
      clearInterval(interval);
    };
  }, []);

  return <TimeWrapper>{currentTime.time.toLocaleTimeString()}</TimeWrapper>;
};
