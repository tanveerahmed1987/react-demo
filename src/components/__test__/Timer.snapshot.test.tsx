// components/__tests__/Timer.snapshot.test.tsx
import * as React from "react";
import { Timer } from "../Timer";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Timer />).toJSON();
  expect(tree).toMatchSnapshot();
});
