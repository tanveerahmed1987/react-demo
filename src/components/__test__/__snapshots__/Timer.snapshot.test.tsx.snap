// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdvvaa iVrGFR"
>
  6:47:22 PM
</div>
`;
